### PR TITLE
docs(css): Create easing function pages

### LINKED ISSUE #29923: Create function pages for easing functions
**MDN URL**: https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function

**Related issue**: This is a follow-up to https://github.com/mdn/content/issues/20138.

**Context**: The `<easing-function>` data type page currently includes documentation for the `linear()`, `cubic-bezier()`, and `steps()` functions. As highlighted in this [review comment](https://github.com/mdn/content/pull/29585#pullrequestreview-1700789490), these functions should each have their dedicated pages.

**Tasks**:
- [x] Create a page for `linear()` function.
- [x] Create a page for `cubic-bezier()` function.
- [x] Create a page for `steps()` function.
- [x] Update `<easing-function>` page.
- [x] Update links for these functions in the [Easing functions](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Functions#easing_functions) section on the "CSS value functions" page.

### PR DESCRIPTION
**Pull request by dipikabh**
### Description

Currently, the documentation for `linear()`, `cubic-bezier()`, and `steps()` easing functions exists only within the [`<easing-function>`](https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function) data type page.

This PR:
- Creates individual reference pages for `linear()`, `cubic-bezier()`, and `steps()` functions from the content on the [`<easing-function>`](https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function) page.

- Updates the `<easing-function>` page:
  - Moves the function details from the "Description" section to the respective function pages.
  - Moves relevant examples to the respective function pages.
  - Aligns the "Values" section with the "Formal syntax" section (retains the definition of the function keywords here).

- Similarly, aligns the "Parameters" section on each function page with its "Formal syntax".

- The "Specifications" section is being updated in https://github.com/mdn/browser-compat-data/pull/25041.

> [!NOTE]  
> This is the first iteration of the function pages. Future improvements can include:
> - Adding interactive examples and more code snippets to the "Examples" section
> - Checking the content against the specification to ensure coverage on MDN

### Related issues and pull requests

Fixes https://github.com/mdn/content/issues/29923

### CODE REVIEW

**Review by pepelsbey — COMMENTED** [review]
Nicely done! Thanks :) I have one fix and a few suggestions:

- Quotes inside of alts might not be useful for the target users or even potentially problematic since we encode them.
- The alt descriptions we provide for graphs might not be comprehensible due to their complexity and length. I wonder if we should reconsider the approach and explain them in prose next to images.

Here’s the longest example:

> Graphs of Input progress to Output progress, of which steps(2, jump-start) shows horizontal lines extending 0.5 unit from (0, 0.5) and (0.5, 1), respectively, with empty circles at the origin and (0.5, 0.5); steps(4, jump-end) shows horizontal lines extending 0.25 unit from (0, 0), (0.25, 0.25), (0.5, 0.5), and (0.75, 0.75), respectively, with unfilled circles at (0.25, 0), (0.5, 0.25), and (0.75, 0.5), and a solid circle at (1, 1); steps(5, jump-none) shows horizontal lines extending 0.2 unit from (0, 0), (0.2, 0.25), (0.4, 0.5), (0.6, 0.75), and (0.8, 1), respectively, with unfilled circles at (0.2, 0), (0.4, 0.25), (0.6, 0.5), and (0.8, 0.75); steps(3, jump-both) shows horizontal lines extending 1/3 unit from (0, 0.25), (1/3, 0.5), and (2/3, 0.75),respectively , with a solid circle at (1, 1) and unfilled circles at the origin, (1/3, 0.25), (2/3, 0.5), and (1, 0.75).

Describing this picture:

image

We might also split the picture in multiple parts and describe each individually.

File: files/en-us/web/css/easing-function/cubic-bezier/index.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+![Graph of "Input progress" to "Output progress" showing an "S"-shaped line curving from the origin to (1, 1) with the Bezier control points P1(0.1, 0.6) and P2(0.7, 0.2).](cubic-bezier.svg)
+
+Not all cubic Bézier curves are suitable as easing functions because not all are [mathematical functions](https://en.wikipedia.org/wiki/Function_%28mathematics%29); i.e., curves that for a given x-axis coordinate have zero or one value. With P0 and P3 fixed as defined by CSS, a cubic Bézier curve is a function, and is therefore valid, if and only if the x-axis coordinates of P1 and P2 are both in the `[0, 1]` range.
+
+Cubic Bézier curves with the P1 or P2 ordinate outside the `[0, 1]` range can cause the value to go farther than the final state and then return. In animations, for some properties, such as {{cssxref("left")}} or {{cssxref("right")}}, this creates a kind of "bouncing" effect.
+
+![Graphs of the easing function "cubic-bezier(0.3, 0.2, 0.2, 1.4)", one of which shows the output progress going above 1 starting from a certain input progress, the other shows the output progress reaching and then staying at 1.](cubic-bezier_out_of_range.svg)
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/web/css/easing-function/cubic-bezier/index.md:56]: I like the fact that we get rid of quotes, and I think we're going to look at these alts in a follow-up, so I'll apply this for now and we can revisit 👍🏻

File: files/en-us/web/css/easing-function/cubic-bezier/index.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+However, certain properties will restrict the output if it goes outside an allowable range. For example, a color component greater than `255` or smaller than `0` in {{CSSXref("color_value/rgb", "rgb()")}} will be clipped to the closest allowed value (`255` and `0`, respectively). Some `cubic-bezier()` values exhibit this property.
+
+## Formal syntax
+
+{{csssyntax}}
+
+## Examples
+
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/web/css/easing-function/cubic-bezier/index.md:65]: very nice! Going to commit for the moment

File: files/en-us/web/css/easing-function/steps/index.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+The following image shows the affect of different `<step-position>` values when the jumps occur:
+
+```css
+steps(2, jump-start)  /* Or steps(2, start) */
+steps(4, jump-end)    /* Or steps(4, end) */
+steps(5, jump-none)
+steps(3, jump-both)
+```
+
+![Graphs of "Input progress" to "Output progress", of which "steps(2, jump-start)" shows horizontal lines extending 0.5 unit from (0, 0.5) and (0.5, 1), respectively, with empty circles at the origin and (0.5, 0.5); "steps(4, jump-end)" shows horizontal lines extending 0.25 unit from (0, 0), (0.25, 0.25), (0.5, 0.5), and (0.75, 0.75), respectively, with unfilled circles at (0.25, 0), (0.5, 0.25), and (0.75, 0.5), and a solid circle at (1, 1); "steps(5, jump-none)" shows horizontal lines extending 0.2 unit from (0, 0), (0.2, 0.25), (0.4, 0.5), (0.6, 0.75), and (0.8, 1), respectively, with unfilled circles at (0.2, 0), (0.4, 0.25), (0.6, 0.5), and (0.8, 0.75); "steps(3, jump-both)" shows horizontal lines extending 1/3 unit from (0, 0.25), (1/3, 0.5), and (2/3, 0.75),respectively , with a solid circle at (1, 1) and unfilled circles at the origin, (1/3, 0.25), (2/3, 0.5), and (1, 0.75).](jump.svg)
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/web/css/easing-function/steps/index.md:68]: holy moly 😂

File: files/en-us/web/css/easing-function/cubic-bezier/index.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+/* There must be exactly four parameters */
+cubic-bezier(0.3, 2.1)
+
+/* X coordinates must be in the [0, 1] range */
+cubic-bezier(-1.9, 0.3, -0.2, 2.1)
+```
+
+## Specifications
+
+{{Specifications}}
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/web/css/easing-function/cubic-bezier/index.md:136]: re-read the PR description, so resolving for now

File: files/en-us/web/css/easing-function/linear/index.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+### Parameters
+
+The function accepts two or more of the following values, which represent progress points in the animation timeline:
+
+- {{cssxref("&lt;number&gt;")}}
+
+  - : Represents a point in time along the duration of the animation or transition.
+    At least two values must be specified.
+    The value `0` represents the start of the iteration, and `1` represents the end.
```
Comment (bsmth) [inline files/en-us/web/css/easing-function/linear/index.md:30]:
```suggestion
    The value `0` represents the start of the transition, and `1` represents the end.
```

**Review by bsmth — APPROVED** [review]
Super, I'm leaving a +1 -> @pepelsbey I have two suggestions if you want to take a look 👀

**Review by pepelsbey — APPROVED** [review]
:shipit:

**Comment by dipikabh** [comment]
Thank you both for taking this PR over the finish line! 🙏

**Pull request merged**